lib/pbench/cli/server/report.py

```diff
@@ -182,86 +240,103 @@ def report_cache(tree: CacheManager):
     )
     click.echo(
-        f"  The most recently used cache, {newest_cache_name}, was "
-        f"referenced {humanize.naturaldate(newest)}"
+        "  The most recently used cache was referenced "
+        f"{humanize.naturaldate(newest)}, {newest_cache_name}"
     )
 
 
 def report_sql():
     """Report the SQL table storage statistics"""
+
+    watcher.update("inspecting SQL tables")
     click.echo("SQL storage report:")
-    click.echo("  Table                Rows       Storage")
-    click.echo("  -------------------- ---------- ----------")
+    t_w = 20
+    r_w = 10
+    s_w = 10
+    click.echo(f"  {'Table':<{t_w}} {'Rows':<{r_w}} {'Storage':<{s_w}}")
+    click.echo(f"  {'':-<{t_w}} {'':-<{r_w}} {'':-<{s_w}}")
     for t in inspect(Database.db_session.get_bind()).get_table_names():
-        rows = list(
+        (rows,) = next(
             Database.db_session.execute(statement=text(f"SELECT COUNT(*) FROM {t}"))
-        )[0][0]
-        size = list(
+        )
+        (size,) = next(
             Database.db_session.execute(
                 statement=text("SELECT pg_total_relation_size(:table)"),
                 params={"table": t},
             )
-        )[0][0]
-        click.echo(f"  {t:20} {rows:>10} {humanize.naturalsize(size):>10}")
+        )
+        click.echo(f"  {t:<{t_w}} {rows:>{r_w},d} {humanize.naturalsize(size):>{s_w}}")
 
     if not detailer:
         return
 
-    query = select(IndexMap.root, IndexMap.index)
-    idxes = Database.db_session.execute(query).all()
+    watcher.update("inspecting index map details")
     record_count = 0
     roots = set()
     indices = set()
     root_size = 0
     index_size = 0
-    for idx in idxes:
+    document_count = 0
+    document_size = 0
+    json_size = 0
+    document_int_size = 0
+
+    query = select(IndexMap.root, IndexMap.index, IndexMap.documents)
+    for root, index, documents in Database.db_session.execute(
+        query, execution_options={"stream_results": True}
+    ).yield_per(500):
         record_count += 1
-        roots.add(idx[0])
-        indices.add(idx[1])
-        root_size += len(idx[0])
-        index_size += len(idx[1])
-    unique_root_size = 0
-    unique_index_size = 0
-    for r in roots:
-        unique_root_size += len(r)
-    for i in indices:
-        unique_index_size += len(i)
+        watcher.update(f"({record_count}: {root} -> {index}")
+        roots.add(root)
+        indices.add(index)
+        root_size += len(root)
+        index_size += len(index)
+        json_size += len(json.dumps(documents))
+        for d in documents:
+            document_count += 1
+            document_size += len(d)
+            i = int(d, base=16)
+            document_int_size += (i.bit_length() + 7) / 8
+        unique_root_size = 0
+        unique_index_size = 0
+        unique_root_size += sum(len(r) for r in roots)
+        unique_index_size += sum(len(i) for i in indices)
```

Comment:
Why initialize `unique_root_size` to zero and then add to it...why not just assign the sum _as_ the initialization?  Ditto for `unique_index_size`.

Reply:
Because you "made" me change to the `sum` and I neglected to remove the pre-loop initializer. Ha.